source/extensions/filters/network/thrift_proxy/header_transport_impl.cc

```diff
@@ -0,0 +1,316 @@
+#include "extensions/filters/network/thrift_proxy/header_transport_impl.h"
+
+#include <limits>
+
+#include "envoy/common/exception.h"
+
+#include "common/buffer/buffer_impl.h"
+
+#include "extensions/filters/network/thrift_proxy/buffer_helper.h"
+#include "extensions/filters/network/thrift_proxy/transport_impl.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace ThriftProxy {
+namespace {
+
+// c.f.
+// https://github.com/apache/thrift/blob/master/lib/cpp/src/thrift/protocol/TProtocolTypes.h#L27
+enum class HeaderProtocolType {
+  Binary = 0,
+  JSON = 1,
+  Compact = 2,
+
+  FirstHeaderProtocolType = Binary,
+  LastHeaderProtocolType = Compact,
+};
+
+// Fixed portion of frame header:
+//   Header magic: 2 bytes +
+//   Flags: 2 bytes +
+//   Sequence number: 4 bytes
+//   Header data size: 2 bytes
+constexpr uint64_t MIN_FRAME_START_SIZE_NO_HEADERS = 10;
```

Comment:
nit: I think we are doing CamelCase now for constants. `MinFrameStartSizeNoHeaders`